shared/src/test/diff/pretyper/ucs/examples/ULC.mls

```diff
@@ -320,20 +320,46 @@ fun search(f: 'A -> Option['B], xs: List['A]): Option['B] =
 //│   where
 //│     'B <: 'A0
 
-// The removal of type annotations will cause running out of fuel.
+// =============================================================================
+// TO BE INVESTIGATED: The following version does not have a stable type.
```

Comment:
Does the type oscillate between different possible results? While the desugared term doesn't change?

Reply:
Yes. Just hit <kbd>Command + S</kbd> to see it changes.

Reply:
It switches between the two types below.

```diff
-//│ fun combinations: forall 'T 'T0 'A. (n: Int, acc: List['T0], alphabet: List['T], xs: List[Str]) -> Option[Str]
+//│ fun combinations: forall 'T 'A 'T0. (n: Int, acc: List['T], alphabet: List['T0], xs: List[Str]) -> Option[Str]
 //│   where
-//│     'T <: 'T0 & 'A
-//│     'T0 :> 'A
-//│     'A := 'T0
+//│     'T0 <: 'T & 'A
+//│     'T :> 'A
+//│     'A := 'T
```

Reply:
It also oscillates if I remove type annotations. Moreover, if I continue to remove the type annotations of some functions above (like `reverse`), the fuel will run out, but I'm not sure which functions to remove.

Reply:
Here is a slightly simpler code whose type also oscillates. I'm sure it has a stable desugared term.

```
fun combinations(n: Int, acc: List['T], alphabet: List['T], xs: List[Str]): Option[Str] =
  if n <= 0 then
    let x = reverse(acc) |> join("")
    if contains(xs, x) then None else Some(x)
  else
    search((x) => combinations(n - 1, x :: acc, alphabet, xs), alphabet)
```

Here is a video showing how it changed.

https://github.com/hkust-taco/mlscript/assets/2239547/c45a11b5-074d-4a89-9f6c-c5fcd162a40b